OpenRA.Mods.Common/TraitsInterfaces.cs

```diff
@@ -354,7 +354,7 @@ public interface IMove
 
 	public interface IRadarSignature
 	{
-		IEnumerable<Pair<CPos, Color>> RadarSignatureCells(Actor self);
+		void RadarSignatureCells(Actor self, List<Pair<CPos, Color>> destinationBuffer);
```

Comment:
`ref` please on `destinationBuffer` to explicitly show that it's going to get modified.

Reply:
`ref` defeats the purpose here. I definitely *don't* want anybody returning a different list! They need to use the one provided I'm happy to use a more verbose name (e.g. `destinationBufferForCells`) if you think that would help clarify.

Reply:
This is a reference type regardless, so a `ref` here only wins you readability in the sense that it explicitly shows that the value/contents is *definitely being* modified inside that method.

Reply:
Considering that a cell can have only one colour, would it make sense to change `List<Pair<CPos, Color>>` to a `Dictionary<CPos, Color>`?